batavia/builtins/hash.js

```diff
@@ -1,8 +1,8 @@
 var exceptions = require('../core').exceptions
 var type_name = require('../core').type_name
-var types = require('../types')
 
 function hash(args, kwargs) {
+    var types = require('../types')
```

Comment:
Node.js may well recommend that - but have they provided an alternative way to resolve circular dependencies in imports?